scipy/optimize/tests/test__differential_evolution.py

```diff
@@ -259,23 +259,45 @@ def test_best_solution_retrieval(self):
         result = solver.solve()
         assert_equal(result.x, solver.x)
 
+    def test_intermediate_result(self):
+        bounds = [(0, 2), (0, 2)]
+        visited = [False]
+        def callback(intermediate_result):
```

Comment:
Please check the other attributes, too. And besides checking for their presence, I imagine a few identities that should hold. For instance, `fun(intermediate_result.x)` equals `intermediate_result.fun`, and `intermediate_result.nit` is the number of times the callback has been called+/- constant depending on convention. It would also be good to check that stopping with the callback after a certain number of iterations (or similar) is equivalent to stopping due to an iteration limit.

Reply:
I've checked some of the attributes in the intermediate result as requested..


> It would also be good to check that stopping with the callback after a certain number of iterations (or similar) is equivalent to stopping due to an iteration limit.

they're equivalent, except `result.message` is different. This is already checked (albeit in separate places) by the test suite.